### PR TITLE
Addresses issues with CRD and Azure provider

### PR DESCRIPTION
**Pull request by sabre1041**
Resolves issue due to `controller-gen` along with additional options on mapping username for Azure provider

### CODE REVIEW

File: config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.4.0
+    controller-gen.kubebuilder.io/version: v0.3.0
```
Comment (raffaelespazzoli) [inline config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml:5]: curious, why did you have to do this?

**Review by raffaelespazzoli — APPROVED** [review]
just a question, otherwise looks good.

File: config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml (hunk 1 above)
Comment (sabre1041) [reply config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml:5]: Errors where returned in .status.Conditions in `v0.4.0`. Need to investigate it later

Comment (raffaelespazzoli) [reply config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml:5]: FYI: I finally got around to managing status conditions in better way in operator-utils.

Comment (davidkarlsen) [reply config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml:5]: @raffaelespazzoli any pointer to that? I have a controller which does something similar

Comment (sabre1041) [reply config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml:5]: @davidkarlsen https://github.com/redhat-cop/operator-utils/

**Pull request merged**